contracts/examples/bonding-curve/src/lib.rs

```diff
@@ -0,0 +1,325 @@
+#![no_std]
+#![allow(unused_attributes)]
+
+elrond_wasm::imports!();
+elrond_wasm::derive_imports!();
+
+mod bc_function;
+use bc_function::{BCFunction, CurveArguments, SupplyType, Token};
+
+mod linear_function;
+use linear_function::LinearFunction;
+
+const TOKEN_NUM_DECIMALS: usize = 18;
+
+#[elrond_wasm_derive::contract]
+pub trait BondingCurve {
+	#[init]
+	fn init(&self, issued_token: Token, exchanging_token: Token) {
```

Comment:
It's a bit difficult to work with serialized structs when providing arguments. I suggest have 4 arguments in total, one for each struct field, and create the Token struct instance yourself from the parts.

Also, please check if the identifiers are valid, by calling is_valid_esdt_identifier().